### ISSUE TITLE
Select Component Options don't change if is multiple

### ISSUE THREAD

**Issue by ismaail** [laravel-filament/filament#13510]
### Package

filament/forms

### Package Version

3.2.92

### Laravel Version

11.14.0

### PHP Version

PHP 8.3.9

### Problem description

In a Product Form, **Variants** are added with a Repeatable Component, 
and this **Variants** are used in  a **Select** Components in another Section in the same Form.

✅ If the **Select** Component is `single`, the changes in the **Variants** Repeatable are reflected in the **Variants** Selects after the form is saved [see video 1](https://github.com/filamentphp/filament/assets/4411670/756eec6c-047f-44ee-b136-ee94064253a5).

❌ But if the **Select** Component is `multiple`, I have to reload the page to see changes in the Select Component [see video 2](https://github.com/filamentphp/filament/assets/4411670/e58b88c1-e0d8-4936-b0bc-78b8283e4649).

_sidenote_: if I remove the `relationship` in the **Select** Component, it works as expected.

### Expected behavior

Select Options to change if a **Variant** is added or removed in the Form.

### Steps to reproduce

1. clone the [repo](https://github.com/ismaail/filament_issue_2)
2. create new Product,
3. Add/Delete Variants and see if Select Component Options change without reloading the page 

### Reproduction repository (issue will be closed if this is not valid)

https://github.com/ismaail/filament_issue_2

**Comment by zepfietje** [comment]
This is caused by the Choices.js select which is used with `multiple()`.
Unfortunately, I don't believe there's anything we can do here.

**Comment by ismaail** [comment]
but if you comment out the **relationship** in `stocks_repeater` Repeater Component, it works just fine [see video 3](https://github.com/user-attachments/assets/df81c333-42cf-4359-a3e1-0fba0545b09d).


```php
->schema([
    Forms\Components\Repeater::make('stocks_repeater')
        ->relationship('stocks')
        ->hiddenLabel()
        ->itemLabel(fn(array $state): ?string => $state['title'] ?? null)
        ->collapsible()
        ->grid(2)
        ->schema([
            Forms\Components\TextInput::make('quantity')->required(),
            Forms\Components\Select::make('variants')
                /*->relationship(
                    name: 'variants',
                    titleAttribute: 'title',
                    modifyQueryUsing: function (Builder $builder, Stock $record) {
                        $builder->where('product_id', '=', $record->product_id);
                    },
                )*/
                ->options(function (Stock $record) {
                    return Variant
                        ::where('product_id', '=', $record->product_id)
                        ->pluck('title', 'id');
                })
                ->multiple()
            ,
        ])
]),
```

**Comment by leandrocfe** [comment]
Could you try to add this?

```php
Forms\Components\Select::make('variants')
...
->extraInputAttributes(['wire:key' => Str::random(10)])
```

**Comment by ismaail** [comment]
@leandrocfe I tried `->extraInputAttributes`  and it didn't work,

then I checked for other similar methods, 
I tried `->extraAlpineAttributes(['wire:key' => Str::random(10)])` and it works.

**Comment by leandrocfe** [comment]
Did you inspect the browser to see if it appears using extraInputAttributes? This was supposed to work

**Comment by leandrocfe** [comment]
https://github.com/filamentphp/filament/discussions/12074#discussioncomment-9168846

**Comment by ismaail** [comment]
yes, with `extraInputAttributes` the `<select>` has the attribute `wire:key`

![image](https://github.com/user-attachments/assets/197ec347-b824-4973-a0f3-6b4f0fa4e5a5)

**Comment by JakeOcean** [comment]
Somewhat related, when using the `createOptionForm` to add a new Option to a Select within a Modal (during Import in my case), the newly created Option never gets selected unless `->searchable()` or `->extraInputAttributes(['wire:key' => Str::random(10)])` is added to the Select

**Comment by adnan-smlatic** [comment]
@ismaail Thank you for this

`->extraAlpineAttributes(['wire:key' => Str::random(10)]`

I couldn't get `disableOptionWhen` in the Select form to update when the options changed, this is an ugly but temporary ok workaround.

**Comment by danharrin** [comment]
Hi all, I think this issue has been fixed in v4 of the framework, so I am closing it. I ended up removing Choices.js and building a custom select where this works.

I will still accept PRs to fix this in older versions as long as bug fixes are supported in the version policy: https://filamentphp.com/docs/4.x/introduction/version-support-policy

Many thanks if you do choose to backport a fix to a previous version!
